frame/lending/src/lib.rs

```diff
@@ -1773,4 +1775,27 @@ pub mod pallet {
 		pub(crate) accrued_increment: T::Balance,
 		pub(crate) new_borrow_index: FixedU128,
 	}
+
+	/// Retrieve the current interest rate for the given `market_id`.
+	pub fn current_interest_rate<T: Config>(market_id: MarketId) -> Result<Rate, DispatchError> {
+		let market_id = MarketIndex::new(market_id);
+		let total_borrowed_from_market_excluding_interest =
+			Pallet::<T>::total_borrowed_from_market_excluding_interest(&market_id)?;
+		let total_available_to_be_borrowed =
+			Pallet::<T>::total_available_to_be_borrowed(&market_id)?;
+		let utilization_ratio = Pallet::<T>::calculate_utilization_ratio(
+			total_available_to_be_borrowed,
+			total_borrowed_from_market_excluding_interest,
+		)?;
+
+		Markets::<T>::try_mutate(market_id, |market_config| {
```

Comment:
Why do we try mutate? If get borrow rate is mut then the storage will be mutated? Perhaps just try_get and mutate after, unless I missee something 🤔

Reply:
Yes, you are right, we can use `try_get` here. Fixed.